cmd/sync/sync.go

```diff
@@ -60,11 +62,21 @@ func (f *syncFlags) syncOptionsFromArgs(cmd *cobra.Command, args []string) (*syn
 		}
 	}
 
+	ctx := cmd.Context()
+	client := root.WorkspaceClient(ctx)
+
+	localRoot := vfs.MustNew(args[0])
+	info, err := git.FetchRepositoryInfo(ctx, localRoot, client)
```

Comment:
What are the failure modes of this?

Before, this would only look for a `.git` folder in ancestors and default to `localRoot` if it couldn't find one.

This means it couldn't fail.

Reply:
Great question. Indeed, this changes how errors are handled.

I've updated the code to match previous situation. Now the errors are logged. We also make use of partial results, like before (got git root but not other stuff) and default to input path in the worst case.

Copy pasting from new comment:

```
// Fetch repository information either by quering .git or by fetching it from API (for dabs-in-workspace case).
//   - In case we could not find git repository, all string fields of GitRepositoryInfo will be "" and err will be nil.
//   - If there were any errors when trying to determine git root (e.g. API call returned an error or there were permission issues
//     reading the file system), all strings fields of GitRepositoryInfo will be "" and err will be non-nil.
//   - For convenience, GuessedWorktreeRoot parameter will be set to path in the above two cases.
//   - If we could determine git worktree root but there were errors when reading metadata (origin, branch, commit), those errors
//     will be logged as warnings, GitRepositoryInfo is guaranteed to have non-empty WorktreeRoot and corresponding GuessedWorktreeRoot
//     and other fields on best effort basis. The err will be nil.
//   - In successful case, all fields are set to proper git repository metadata.
func FetchRepositoryInfo(ctx context.Context, path vfs.Path, w *databricks.WorkspaceClient) (GitRepositoryInfo, error) {
```